Add workspace icons

barklund opened the pull request:
## Context

See #6204 for todo.

## Testing Instructions

### QA

This PR can be tested by following these steps:

1. Load up a story
2. Observe the new icons in the page navigation, page menu, carousel, and carousel menu
3. Verify that all the above-mentioned icons work as intended (as before)

### UAT

<!-- ignore-task-list-start -->
- [x] UAT should use the same steps as above.
<!-- ignore-task-list-end -->

## Checklist

<!-- Check these after PR creation -->

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6204

Commit:
Added new arrows using design system

Commit:
Using existing deepmerge lib, but not actively adding to package.json

This is only temporary until we come up with the proper solution in #6209

barklund commented on assets/src/edit-story/theme.js:
```diff
@@ -80,173 +85,176 @@ export function useTheme(selector) {
-      white: '#FFFFFF',
-      black: '#000000',
-      divider: '#FFFFFF',
+const theme = deepmerge(
```
Note that this is a temporary workaround until #6209 solves this correctly.

barklund replied:
Now removed

barklund commented on assets/src/edit-story/theme.js:
```diff
@@ -19,19 +19,24 @@
  */
 import { createGlobalStyle, ThemeContext } from 'styled-components';
 import { rgba } from 'polished';
+import deepmerge from 'deepmerge';
```
We have this through some other dependency, so it builds, but we don't directly depend on it, so it doesn't lint.

swissspidy replied:
It would need to be added to `package.json` explicitly

barklund replied:
It's temporary only until #6209 is merged in, that's why I didn't add it. I just needed it to test locally.

barklund replied:
Now removed again.

Commit:
Added new arrow buttons and spacing to carousel

Commit:
Changed buttons to circles because I think that is the intent?

Commit:
Updated to new icon buttons in the page menu

They do not look very good though. Will have to ask about that.

Commit:
Added page menu icons

Had to change some things around the page menu layout as well, but some of those overlaps with zoom PR i #4022, so will have some conflicts there.

Commit:
Generalized common props

barklund commented on assets/src/edit-story/components/canvas/pagemenu/index.js:
```diff
@@ -249,40 +239,29 @@ function PageMenu() {
             </WithTooltip>
           )}
         </Options>
-        {showTextMagicAndHelperMode && (
-          <Options>
-            <Icon disabled>
-              <LayoutHelper />
-            </Icon>
-            <Space isDouble />
-            <Icon disabled>
-              <Text />
-            </Icon>
-          </Options>
-        )}
```
I decided to fully remove this (but not the underlying flag) as it was just some old icons lying around with no accompanying functionality. If we ever need this functionality, design has to be redone anyway.

@swissspidy, should I remove the flag as well?

swissspidy replied:
Yeah let's also remove the flag.

barklund replied:
Removed in 0a21ace.

barklund commented on assets/src/edit-story/components/canvas/layout.js:
```diff
@@ -47,7 +47,7 @@ export const Z_INDEX = {
 };
 
-const MENU_HEIGHT = 48;
+const MENU_HEIGHT = 52;
```
This (random magic number) will all be done [properly](https://github.com/google/web-stories-wp/pull/6236/files#diff-f634fcb15355d7101d454a9ad3b2aa06ceb1ce6bd5ebd0f4e8dcf55c9ded3441R59) in the zoom PR in #6236 , so ignore for this one.

barklund commented on assets/src/edit-story/components/canvas/pagemenu/index.js:
```diff
@@ -77,35 +73,30 @@ const Options = styled.div`
   background-color: ${({ theme }) => theme.colors.fg.white};
   opacity: 0.3;
-  height: ${HEIGHT}px;
+  height: ${HEIGHT / 2}px;
```
So the icons happen to be half the height of the buttons and the divider should follow the icon height... Is there a cleaner way to do this?

miina replied:
Maybe an option could be adding the sizes of the button variants in pixels as well to the `button/index.js` in the design-system folder and using that instead of height? Would be great if we could still access the size in px somehow as well for these cases specifically -- when other things depend on the size, might not be the only time when this happens. Then, inside the `button/index.js` also these constants could be used throughout the file instead of using numbers directly. Thoughts?

barklund replied:
So, I added the pixel sizes of icon buttons to the button constants in the design system. I'm not loving it, but it works okay. Naming could be optimized.

I'd like input from @BrittanyIRL as well here.

* Adding constants: dda29454e
* Using them: 0fa5c1cc4

barklund replied:
(also, the icons weren't actually half the size - but slightly smaller than that even: 14px vs 32px button)

BrittanyIRL replied:
I feel ok about this because we're already defining the height in pixels here. I would say 1 nit would be can we name the  group `BUTTON_HEIGHT_PIXELS` or something similar that can give us a bit of context as to what the pixels are? 

My assumption reading through this is that the divider would always want to be relative to the button sizes so it makes sense to tie it together.

barklund replied:
Well, it is both the width and the height of the elements (and in other places we're using it to express width), so that's why the dimension-neutral name. And `SIZE` was already taken.

But it could perhaps be `BUTTON_DIMENSIONS` instead?

barklund replied:
Changed all this completely, as icons only have one size now. There's a theme constant for the large button size and another for the icon size. That's it.

barklund replied:
(and neither is used for this particular variable - it's completely custom set to 16px, which looks good, but doesn't directly match the icon heights, as they vary a slight bit)

miina commented on assets/src/edit-story/components/canvas/layout.js:
```diff
@@ -123,7 +123,7 @@ const PageAreaSafeZone = styled.div`
 const HeadArea = styled(Area).attrs({ area: 'head', overflowAllowed: false })``;
 
-const MenuArea = styled(Area).attrs({ area: 'menu', overflowAllowed: false })``;
+const MenuArea = styled(Area).attrs({ area: 'menu', overflowAllowed: true })``;
```
Hm, what's the overflow for?

barklund replied:
The focus border overflows. Without it you get:

![Screen Shot 2021-02-09 at 17 22 01](https://user-images.githubusercontent.com/637548/107436341-647a3c00-6afb-11eb-84d2-b5cbf2edce09.png)

Which looks stupid 🤷 . And I don't think the overflow here hurts.

miina commented on assets/src/edit-story/components/carousel/carouselScroll.js:
```diff
@@ -19,15 +19,27 @@
 
-function CarouselScrollButton({ isForward = false }) {
+const FlippableArrow = styled(Icons.Arrow)`
+  transform-origin: center center;
```
Isn't this the default?

barklund replied:
Indeed! Fixed in 0000c9331 .

miina commented on assets/src/edit-story/components/carousel/carouselLayout.js:
```diff
@@ -27,19 +27,21 @@ import { CarouselScrollForward, CarouselScrollBack } from './carouselScroll';
 import CarouselList from './carouselList';
 import useCarousel from './useCarousel';
-import { MENU_GUTTER, BUTTON_GUTTER } from './constants';
+import { MENU_GUTTER, BUTTON_WIDTH, BUTTON_GAP } from './constants';
```
Here, too -- there are two different general button sizes (not only related to the carousel) that seem to be needed elsewhere, too, so this should probably come from the design system (`buttons/index.js` or separate constants file) instead.

barklund replied:
Referring to the [previous discussion](https://github.com/google/web-stories-wp/pull/6272#discussion_r573286458).

miina commented on assets/src/edit-story/components/canvas/pagenav/index.js:
```diff
@@ -68,6 +58,9 @@ function PageNav({ isNext }) {
   const { isRTL } = useConfig();
+
+  // Cancel lasso on mouse down
+  const cancelMouseDown = useCallback((evt) => evt.stopPropagation(), []);
```
Tried to test on `main` to understand what was happening before but wasn't sure. Curious what was the previous behavior that this prevents now for the user.

barklund replied:
It was also there [before](https://github.com/google/web-stories-wp/pull/6272/files/ac66669804e66fce95eeda6eddbfe9059ee26739#diff-02139befcc819b7d91d045078dd199c489e48c9dce377d439faf7b0dbcdc8c52L90) - I just memoized it.

miina commented on assets/src/edit-story/components/canvas/pagenav/index.js:
```diff
@@ -17,49 +17,39 @@
-const RightNavButton = styled(RightArrow)`
-  ${pageNavButtonsStyle}
+const FlippableArrow = styled(Icons.Arrow)`
+  transform-origin: center center;
```
Here, too, isn't this the default?

barklund replied:
Fixed in 0000c9331.

Commit:
Merge branch 'main' into add/6204-workspace-icons

Commit:
Removed unused flag

Commit:
Added new button pixels constants

Commit:
Employed new button pixel constants

Commit:
Added one more case of button pixel usage

Commit:
Changed all buttons to square (never circles!)

Commit:
Removed irrelevant css property

Commit:
Much better grid structure!

Commit:
Moved dashboard specific icons out of design system

Commit:
Replaced all icons with new ones

This has many consequences as the new icons are differently boxed compared to the old ones. I will try to fix in subsequent commits.

Commit:
Fixed icon styling

Commit:
Added small icons, fixed dropdown chevron

Commit:
Fixed cross icon in snackbar

Commit:
Changed all icon colors to dynamic and added storybook demo

Commit:
Fixed inspector icon usage and storybook

Commit:
Fixed plus icon in color picker

Commit:
Fixed xs chevron icons after alignment

Commit:
Fixed plus icons in color preset empty panel

Commit:
Fixed icons in preset panel header

Commit:
Swapped two icons

Commit:
Fixed design in help center including new icons

Commit:
Reversed help center chevron direction

Commit:
Renamed icon

Commit:
Merge branch 'main' into task/6299-ds-icons

Commit:
Updated instructions

BrittanyIRL committed:
Made some icon adjustments

Make checkmark bigger in design system (used on dashboard/settings and telemetry banner), make toggle checkmark bigger (only exists in storybook), tweak prepublish alert icon that shows up on publisher or update button to be better size, move space for help center toggle to be padding instead of a div. (#6319)

Commit:
Edited icon instructions

Commit:
Merge branch 'task/6299-ds-icons' of github.com:google/web-stories-wp into task/6299-ds-icons

Commit:
Merge branch 'task/6299-ds-icons' into add/6204-workspace-icons

# Conflicts:
#	assets/src/design-system/components/button/index.js
#	assets/src/edit-story/components/canvas/pagemenu/index.js
#	assets/src/edit-story/components/canvas/pagenav/index.js
#	assets/src/edit-story/theme.js

Commit:
Fixed page menu sizing and structure

Commit:
Refactored page menu

Commit:
Fixed page count spacing

Commit:
Updated page count to reflect recent UX review

Commit:
Fixed some icon colors

Commit:
Fixed some icon colors

Commit:
Merge branch 'task/6299-ds-icons' into add/6204-workspace-icons

swissspidy commented on assets/src/design-system/icons/chevron_up_small.svg:
```diff
@@ -1,3 +1,3 @@
 <svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
-<path d="M12 17L16 14L20 17" stroke="#E4E5E6" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
+<path d="M12 17L16 14L20 17" stroke="currentColor" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
```
FWIW our SVGO config should be doing this automatically under the hood.

Good practice nontheless 👍

barklund replied:
Might this be missing for Storybook only then? Because without it, the icons clearly weren't colorable (in Storybook)) - which is why I noticed these exact problems.

swissspidy replied:
Strange.

We definitely do load svgr in our storybook webpack config:

https://github.com/google/web-stories-wp/blob/38e3513c7e8ba1d46c5bde5d5572c662d89803cc/.storybook/main.cjs#L52

If that's broken, would you mind opening a ticket for us to investigate? Thanks!

barklund replied:
I can confirm it doesn't work in Storybook, only in the actual editor build. Maybe it's because `svgo.yml` is in the root directory, so it's not picked up inside the `storybook` folder? Follow-up ticket in #6357.

miina commented on assets/src/edit-story/components/canvas/pagemenu/pageMenuButton.js:
```diff
@@ -0,0 +1,54 @@
+
+function PageMenuButton({ children, title, shortcut, ...rest }) {
+  return (
+    <WithTooltip title={title} shortcut={shortcut}>
```
Use `Tooltip` from the `design-system` instead?

barklund replied:
Yes, we should implement that too. But let's hold off on doing it piecemeal and implement it through somewhere. It's probably easiest to change the insides of `WithTooltip` instead actually. Noted for future ticket 📝

miina commented on assets/src/edit-story/components/canvas/pagemenu/pageMenu.js:
```diff
@@ -0,0 +1,186 @@
+        onClick={handleRedo}
+        aria-label={__('Redo Changes', 'web-stories')}
+      >
+        {isRTL ? <Icons.ArrowDownleftCurved /> : <Icons.ArrowDownrightCurved />}
```
Nit: Should `left` and `right` be with capital letters, too? `ArrowDownLeftCurved` etc.

barklund replied:
Done in [the icon PR](https://github.com/google/web-stories-wp/pull/6311/commits/4b24d3ce8c32d41a1c6daac2ef80bd024c8ea67d).

miina commented on assets/src/edit-story/components/canvas/pagemenu/pageMenu.js:
```diff
@@ -0,0 +1,186 @@
+    [addPage, currentPage]
+  );
+
+  const isWidePage = pageSize.width > 280;
```
280 to a constant perhaps?

barklund replied:
It's temporary only until zoom is merged in - then this will change to some actual calculated value.

miina commented on assets/src/edit-story/components/canvas/pagemenu/pageMenu.js:
```diff
@@ -0,0 +1,186 @@
+const PageCount = styled.div`
+  color: ${({ theme }) => theme.colors.fg.primary};
+  font-family: ${({ theme }) => theme.DEPRECATED_THEME.fonts.body1.family};
+  font-size: ${({ theme }) => theme.DEPRECATED_THEME.fonts.body1.size};
```
Should these be moved to the design-system under `typography`, too?

barklund replied:
I could do [even better actually](https://github.com/google/web-stories-wp/pull/6272/files#diff-e12246607f5678a452584b8ec9cfc1990d830a3db193f87d7bf2b385ed760b0dR114-R120)! Thanks for pushing me when I'm lazy 😉 .

miina commented on assets/src/edit-story/components/canvas/layout.js:
```diff
@@ -63,26 +64,45 @@ const Layer = styled.section`
+    '. . . . . . .' ${MENU_GAP}px
+    'm m m m m m m' ${MENU_HEIGHT}px
+    '. . . . . . .' 1fr
+    'c c c c c c c' ${CAROUSEL_HEIGHT}px
```
Yay! Almost looks like a room full of (lined up) monsters in [Nethack](https://en.wikipedia.org/wiki/NetHack) 😍 .

Commit:
Fixed icon size in storybook

Commit:
Merge branch 'main' into task/6299-ds-icons

Commit:
Renamed some icons

Commit:
Fixed position and css for ds toggle checkmark

Commit:
Scaled page layouts icon back up

Commit:
Updated styling of + in color picker text

Commit:
Updated help center toggle button layout

Commit:
Exit debug

Commit:
Merge branch 'task/6299-ds-icons' into add/6204-workspace-icons

Commit:
Used existed typography

miina reviewed (approved):
LGTM Pending the icons PR merge

Commit:
Merge branch 'main' into add/6204-workspace-icons

# Conflicts:
#	assets/src/design-system/components/button/index.js

Commit:
Merge branch 'main' into add/6204-workspace-icons

# Conflicts:
#	assets/src/design-system/components/button/constants.js
#	assets/src/edit-story/components/canvas/pagemenu/index.js
#	assets/src/edit-story/components/carousel/carouselMenu.js
